src/test/java/org/javarosa/core/model/actions/OdkNewRepeatEventTest.java

```diff
@@ -0,0 +1,86 @@
+package org.javarosa.core.model.actions;
+
+import org.javarosa.core.test.Scenario;
+import org.javarosa.xform.parse.XFormParseException;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.isEmptyOrNullString;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static org.javarosa.test.utils.ResourcePathHelper.r;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Specification: https://opendatakit.github.io/xforms-spec/#the-odk-new-repeat-event.
+ */
+public class OdkNewRepeatEventTest {
+    @Test
+    public void setValueOnRepeatInsertInBody_setsValueInRepeat() {
+        Scenario scenario = Scenario.init(r("event-odk-new-repeat.xml"));
+
+        assertThat(scenario.repeatInstancesOf("/data/my-repeat").size(), is(0));
+        scenario.createMissingRepeats("/data/my-repeat[0]");
+        assertThat(scenario.repeatInstancesOf("/data/my-repeat").size(), is(1));
+        assertThat(scenario.answerOf("/data/my-repeat[0]/defaults-to-now"), is(notNullValue()));
+    }
+
+    @Test
```

Comment:
This one could look like this:
```java
@Test
public void addingRepeat_doesNotChangeValueSetForPreviousRepeat() {
    Scenario scenario = Scenario.init(r("event-odk-new-repeat.xml"));
    scenario.next();
    scenario.next();
    scenario.createNewRepeat();

    String firstRepeatDisplayText = scenario.answerOf("/data/my-repeat[0]/defaults-to-now").getDisplayText();
    // Go through the couple of fields in the group
    scenario.next();
    scenario.next();
    // Exit the group
    scenario.next();

    scenario.createNewRepeat();
    String secondRepeatDisplayText = scenario.answerOf("/data/my-repeat[1]/defaults-to-now").getDisplayText();

    assertThat(scenario.answerOf("/data/my-repeat[0]/defaults-to-now").getDisplayText(), is(firstRepeatDisplayText));
    assertThat(secondRepeatDisplayText, is(not(firstRepeatDisplayText)));
}
```

Reply:
Having narrow-focused forms, the number of calls to `scenario.next()` would decrease.